src/main/java/com/klodnicki/Bike/v2/service/BikeService.java

```diff
@@ -77,50 +78,26 @@ public void deleteById(Long id) {
 
-        Bike bike = bikeRepository.findById(id).orElseThrow(IllegalAccessError::new);
-
-        if (updatedBikeRequestDTO.getSerialNumber() != null) {
-            bike.setSerialNumber(updatedBikeRequestDTO.getSerialNumber());
-        }
+        Bike bike = bikeRepository.findById(id).orElseThrow(IllegalArgumentException::new);
 
+        updateBikeIfValuesAreNotNulls(updatedBikeRequestDTO, bike);
         bike.setRented(updatedBikeRequestDTO.isRented());
 
-        if (updatedBikeRequestDTO.getBikeType() != null) {
-            bike.setBikeType(updatedBikeRequestDTO.getBikeType());
-        }
-        if (updatedBikeRequestDTO.getRentalStartTime() != null) {
-            bike.setRentalStartTime(updatedBikeRequestDTO.getRentalStartTime());
-        }
-        if (updatedBikeRequestDTO.getRentalEndTime() != null) {
-            bike.setRentalEndTime(updatedBikeRequestDTO.getRentalEndTime());
-        }
-        if (updatedBikeRequestDTO.getAmountToBePaid() != 0) {
-            bike.setAmountToBePaid(updatedBikeRequestDTO.getAmountToBePaid());
-        }
-        if (updatedBikeRequestDTO.getGpsCoordinates() != null) {
-            bike.setGpsCoordinates(updatedBikeRequestDTO.getGpsCoordinates());
-        }
-
         bikeRepository.save(bike);
 
-        //converting Bike into BikeForAdminResponseDTO using builder
-        return convertBikeIntoBikeForAdminResponseDTO(bike);
+        //converting Bike into BikeForAdminResponseDTO using model mapper
+        return modelMapper.map(bike, BikeForAdminResponseDTO.class);
     }
 
-    private static BikeForAdminResponseDTO convertBikeIntoBikeForAdminResponseDTO(Bike savedBike) {
-        // Builder is in Lombok dependency
-        // BikeForAdminResponseDTO must have @Builder annotation
-        BikeForAdminResponseDTO bikeDTO = BikeForAdminResponseDTO.builder()
-                .id(savedBike.getId())
-                .serialNumber(savedBike.getSerialNumber())
-                .isRented(savedBike.isRented())
-                .bikeType(savedBike.getBikeType())
-                .rentalStartTime(savedBike.getRentalStartTime())
-                .rentalEndTime(savedBike.getRentalEndTime())
-                .amountToBePaid(savedBike.getAmountToBePaid())
-                .gpsCoordinates(savedBike.getGpsCoordinates())
-                .build();
-        return bikeDTO;
-    }
+    private static void updateBikeIfValuesAreNotNulls(BikeRequestDTO updatedBikeRequestDTO, Bike bike) {
 
+        Optional.ofNullable(updatedBikeRequestDTO.getSerialNumber()).ifPresent(bike::setSerialNumber);
+        Optional.ofNullable(updatedBikeRequestDTO.getBikeType()).ifPresent(bike::setBikeType);
+        Optional.ofNullable(updatedBikeRequestDTO.getRentalStartTime()).ifPresent(bike::setRentalStartTime);
+        Optional.ofNullable(updatedBikeRequestDTO.getRentalEndTime()).ifPresent(bike::setRentalEndTime);
+        if (updatedBikeRequestDTO.getAmountToBePaid() != 0) {
```

Comment:
That if statement I pointed out earlier is still hiding among now other Optionals and it shouldn't be found here!

Reply:
okay, then I'm replacing this condition putting it to update method.